pkg/functions/openfaas/driver.go

```diff
@@ -94,6 +122,25 @@ func (d *ofDriver) Create(f *functions.Function, exec *functions.Exec) error {
 		}
 		return errors.Wrapf(err, "Error performing POST request, status: %v", res.StatusCode)
 	}
+
+	// make sure the function has started
+	for i := 0; i < 10; i++ {
```

Comment:
Is 10 seconds the right timeout?

Reply:
Fixing it right now... I'm replacing it with exponential backoff with a total max wait of about 1 minute. Would that work?

Reply:
Yeah, I think that's good.  Just for kicks, let's make it configurable (max timeout).